subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/NodeState.java

```diff
@@ -944,6 +944,7 @@ private void removeOutgoingEdges() {
         }
         virtualEdges = null;
         previousTraversalExclusions = null;
+        cachedFilteredDependencyStates = null;
```

Comment:
I'm afraid this will have an impact on performance on the exclude rule merging perf test. Let's see.

Reply:
Could not show any impact on performance, so I believe we are good. Which I believe makes sense as a node being deselected then re-selected is pretty rare.